File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.go
import (
    "sort"
)

type Time struct {
    value int
    start bool
}

func minMeetingRooms(intervals [][]int) int {
    var times []Time
    numberOfOngoingMeetings := 0
    maximumNumberOfMeetings := 0

    for _, interval := range intervals {
        times = append(times, Time{interval[0], true})
        times = append(times, Time{interval[1], false})
    }    

    sort.Slice(times, func(i, j int) bool {
        if times[i].value == times[j].value {
            return !times[i].start && times[j].start
        }
        return times[i].value < times[j].value
    })

    for _, time := range times {
        if time.start {
            numberOfOngoingMeetings++
        } else {
            numberOfOngoingMeetings--
        }

        maximumNumberOfMeetings = max(maximumNumberOfMeetings, numberOfOngoingMeetings)
    }

    return maximumNumberOfMeetings
}

File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        times = []
        numberOfOngoingMeetings = 0
        maximumNumberOfMeetings = 0
        
        for interval in intervals:
            times.append([interval[0], "start"])
            times.append([interval[1], "end"])

        times.sort()
        
        for time in times:
            if time[1] == "start":
                numberOfOngoingMeetings += 1
            else:
                numberOfOngoingMeetings -= 1
            maximumNumberOfMeetings = max(maximumNumberOfMeetings, numberOfOngoingMeetings)

        return maximumNumberOfMeetings
